registry.go

```diff
@@ -58,6 +59,7 @@ type registryImpl struct {
 	sourceId  string
 	tags      map[string]string
 	metricMap cmap.ConcurrentMap[any]
+	lock      sync.Mutex
```

Comment:
I don't see this used anywhere in this PR?

Reply:
It's used in usageRegistry. I moved it down to the type and did a few other bits of metric initialization cleanup